Structs compilation error(s)

**V version:** 0.1.17
**OS:** ubuntu 19

**What did you do?**
I want to compile this library https://github.com/vpervenditti/vgram

**What did you expect to see?**
a correct compilation
 
**What did you see instead?**
```
.b.c: In function ‘vgram__Bot_get_updates’:
.b.c:9:30: error: expected expression before ‘{’ token
 #define STRUCT_DEFAULT_VALUE {}
                              ^
.b.c:6215:18: note: in expansion of macro ‘STRUCT_DEFAULT_VALUE’
 tmp2 . message = STRUCT_DEFAULT_VALUE;
                  ^~~~~~~~~~~~~~~~~~~~
.b.c:9:30: error: expected expression before ‘{’ token
 #define STRUCT_DEFAULT_VALUE {}
                              ^
.b.c:6216:25: note: in expansion of macro ‘STRUCT_DEFAULT_VALUE’
 tmp2 . edited_message = STRUCT_DEFAULT_VALUE;
                         ^~~~~~~~~~~~~~~~~~~~
.b.c:9:30: error: expected expression before ‘{’ token
 #define STRUCT_DEFAULT_VALUE {}
                              ^
.b.c:6217:23: note: in expansion of macro ‘STRUCT_DEFAULT_VALUE’
 tmp2 . channel_post = STRUCT_DEFAULT_VALUE;
                       ^~~~~~~~~~~~~~~~~~~~
.b.c:9:30: error: expected expression before ‘{’ token
 #define STRUCT_DEFAULT_VALUE {}
                              ^
.b.c:6218:30: note: in expansion of macro ‘STRUCT_DEFAULT_VALUE’
 tmp2 . edited_channel_post = STRUCT_DEFAULT_VALUE;
                              ^~~~~~~~~~~~~~~~~~~~
.b.c:9:30: error: expected expression before ‘{’ token
 #define STRUCT_DEFAULT_VALUE {}
                              ^
.b.c:6219:25: note: in expansion of macro ‘STRUCT_DEFAULT_VALUE’
 tmp2 . callback_query = STRUCT_DEFAULT_VALUE;
                         ^~~~~~~~~~~~~~~~~~~~
.b.c: In function ‘vgram__Bot_send_message’:
.b.c:9:30: error: expected expression before ‘{’ token
 #define STRUCT_DEFAULT_VALUE {}
                              ^
.b.c:6275:16: note: in expansion of macro ‘STRUCT_DEFAULT_VALUE’
 tmp10 . from = STRUCT_DEFAULT_VALUE;
                ^~~~~~~~~~~~~~~~~~~~
.b.c:9:30: error: expected expression before ‘{’ token
 #define STRUCT_DEFAULT_VALUE {}
                              ^
.b.c:6277:16: note: in expansion of macro ‘STRUCT_DEFAULT_VALUE’
 tmp10 . chat = STRUCT_DEFAULT_VALUE;
                ^~~~~~~~~~~~~~~~~~~~
.b.c:9:30: error: expected expression before ‘{’ token
 #define STRUCT_DEFAULT_VALUE {}
                              ^
.b.c:6278:24: note: in expansion of macro ‘STRUCT_DEFAULT_VALUE’
 tmp10 . forward_from = STRUCT_DEFAULT_VALUE;
                        ^~~~~~~~~~~~~~~~~~~~
.b.c:9:30: error: expected expression before ‘{’ token
 #define STRUCT_DEFAULT_VALUE {}
                              ^
.b.c:6279:29: note: in expansion of macro ‘STRUCT_DEFAULT_VALUE’
 tmp10 . forward_from_chat = STRUCT_DEFAULT_VALUE;
                             ^~~~~~~~~~~~~~~~~~~~
.b.c:9:30: error: expected expression before ‘{’ token
 #define STRUCT_DEFAULT_VALUE {}
                              ^
.b.c:6286:17: note: in expansion of macro ‘STRUCT_DEFAULT_VALUE’
 tmp10 . audio = STRUCT_DEFAULT_VALUE;
                 ^~~~~~~~~~~~~~~~~~~~
.b.c:9:30: error: expected expression before ‘{’ token
 #define STRUCT_DEFAULT_VALUE {}
                              ^
.b.c:6289:28: note: in expansion of macro ‘STRUCT_DEFAULT_VALUE’
 tmp10 . left_chat_member = STRUCT_DEFAULT_VALUE;
                            ^~~~~~~~~~~~~~~~~~~~
V panic: C error. This should never happen. Please create a GitHub issue: https://github.com/vlang/v/issues/new/choose

```

--- a/compiler/parser.v
+++ b/compiler/parser.v
@@ -3278,7 +3278,7 @@ fn (p mut Parser) js_decode() string {
 		for field in T.fields {
 			def_val := type_default(field.typ)
 			if def_val != '' {
-				decl += '$tmp . $field.name = $def_val;\n'
+				decl += '$tmp . $field.name = OPTION_CAST($field.typ) $def_val;\n'
 			}
 		}
 		p.gen_json_for_type(T)

